src/main/assets/js/courts-table-search.ts

```diff
@@ -0,0 +1,220 @@
+    const filteredCourts = CourtsTableSearch.filterCourts(courts, searchFilterValue, includeClosedCourts,
+      orderNameAscendingFilter, orderUpdatedAscendingFilter);
+    $(this.courtsResultsSection).html(CourtsTableSearch.getCourtsTableBody(filteredCourts));
+    searchFilterValue.length ? $(this.numberOfCourts).show().text(
+      'Showing ' + filteredCourts.length + ' results')
+      : $(this.numberOfCourts).hide();
+  }
+
+  /**
+   *
+   * Takes a list of courts, and filters based on a set list of criteria:
+   * 1. Name Asc/Desc
+   * 2. Updated date Asc/Desc
+   * 3. Resets whichever one is not enabled
+   * 4. Based on the above, also filters the result list based on the search string from the 'search' input field
+   *
+   * @param courts the list of public courts
+   * @param searchFilterValue the string value of the search input field
+   * @param includeClosedCourts whether or not to include closed courts in the result
+   * @param orderNameAscendingFilter whether to sort by name asc or desc
+   * @param orderUpdatedAscendingFilter whether to sort by last updated date asc or desc
+   * @private
+   */
+  private static filterCourts(courts: string, searchFilterValue: string, includeClosedCourts: boolean,
+    orderNameAscendingFilter: string, orderUpdatedAscendingFilter: string): any {
+
+    return JSON.parse(courts)
+      .filter((court: { displayed: boolean }) => {
+        if (includeClosedCourts) {
+          return court.displayed == true || court.displayed == false;
+        }
+        else return court.displayed == true;
+      }).filter((court: { name: string }) => {
+        if (searchFilterValue.length == 0)
+          return court;
+        else if (court.name.toLowerCase().includes(searchFilterValue.toString().toLowerCase())) {
+          return court;
+        }
+      }).sort(((a: { name: string; updated_at: string }, b: { name: string; updated_at: string }) => {
+
+        switch (orderNameAscendingFilter) {
+          case orderToggleState.ASC:
+            return (a.name > b.name) ? 1 : -1;
+          case orderToggleState.DESC:
+            return (a.name < b.name) ? 1 : -1;
+          default:
+            break;
+        }
+        switch (orderUpdatedAscendingFilter) {
+          case orderToggleState.ASC:
+            return (Date.parse(a.updated_at) < Date.parse(b.updated_at)) ? 1 : -1;
+          case orderToggleState.DESC:
+            return (Date.parse(a.updated_at) > Date.parse(b.updated_at)) ? 1 : -1;
+          default:
+            break;
+        }
+      }));
+  }
+
+  /**
+   *
+   * Create and return the tbody of the table based on a list of table row elements that are created
+   * by looping through each filtered court
+   *
+   * @param filteredCourts the filtered list of public courts
+   * @private
+   */
+  private static getCourtsTableBody(filteredCourts: any): string {
+    let tableData = '';
+    $.each(filteredCourts,function(index,value) {
```

Comment:
You could use an arrow function here. Also, should we specify the types for the function?